### PR TITLE
Tech team safari i os fix

### PR DESCRIPTION
**Pull request by omarcodex**
So this should fix the mobile safari issues, and at the same time finish current thread of `coordinate.js` refactor. 

* Moved user-session dependent calls to authStateChanged;
* Completed refactor of 'the beast' by parsing icon generation into contained code for readability and future changeability;
* Hope it helps!

### CODE REVIEW

**Comment by JohannBillar** [comment]
@omarcodex Do you want to spend an hour today and walk me through the new code and then w'll merge it with master.

File: scripts/coordinate.js [hunk 1]
```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
...
-              // console.log(childSnap.val());
-              // console.log('childsnap.val', childSnap.val());
-              doerUserObjects.push(childSnap.val());
-            }
-          });
-          if (creatorEmail === childSnap.val().email) {
-            creatorUserObjects.push(childSnap.val());
+function renderDoWhopMainHeader(userSessionCurrentDoWhop) {
+  var currentDoWhopID = userSessionCurrentDoWhop; // This is available from higher scope.
+  firebase.database().ref().child('DoWhopDescriptions/' + userSessionCurrentDoWhop).once('value').then(function(data) {
```
Comment (JohannBillar) [inline scripts/coordinate.js]: Could be more readable as: 
database.ref('DoWhopDescriptions).child(currentDoWhopID)
  .once('value')
 .then(function(data) {});

File: scripts/coordinate.js [hunk 2]
```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
...
+        '</div>';
+    });
+  }
+  return textOutput;
+}
+
+function checkDoWhopDetails(userSessionCurrentDoWhop) {
+  var currentDoWhopID = userSessionCurrentDoWhop;
+  // Turning on a listener to the DoWhopDescriptions for a given DoWhop ID:
+  firebase.database().ref().child('DoWhopDescriptions/' + userSessionCurrentDoWhop).on('value', function(data) {
```
Comment (JohannBillar) [inline scripts/coordinate.js]: Could be more readable as:
database.ref('DoWhopDescriptions).child(currentDoWhopID)
  .on('value', function(data) {});

File: scripts/coordinate.js [hunk 3]
```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
...
+    var renderWhoAmIInformation = data.val().whoAmIDescription || '';
+    var renderWhatInformation = data.val().whatDescription;
+    var renderWhenInformation = data.val().whenDescription;
+    var renderWhereInformation = data.val().whereDescription;
+
+    // Checking for updates to renderWhereInformation - DEV refactor.
+    if (data.val().whereAddress && data.val().whereAddress != 'By request') {
+      renderWhereInformation = data.val().whereAddress;
+    }
+
```
Comment (JohannBillar) [inline scripts/coordinate.js]: Would this work without implicit coercion? Eg.
if (
      data.val().whenDateTime && data.val().whenDateTime !== ('By request' || 'tbd')
    ) {}

File: scripts/coordinate.js [hunk 4]
```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
...
+    var doWhopDescriptionTitle = data.val().titleDescription;
+    var renderWhyInformation = data.val().whyDescription;
+    var renderWhenInformation = data.val().whenDescription;
+    var renderWhereInformation = data.val().whereDescription;
+    var renderWhoDescription = data.val().whoDescription;
+    var renderWhoAmIInformation = data.val().whoAmIDescription || '';
+    var renderWhatInformation = data.val().whatDescription;
+    var renderWhenInformation = data.val().whenDescription;
+    var renderWhereInformation = data.val().whereDescription;
+
```
Comment (JohannBillar) [inline scripts/coordinate.js]: Would this work without implicit coercion? Eg.
if (
      data.val().whereAddress && data.val().whereAddress !== 'By request') {
) {}

Comment (omarcodex) [reply scripts/coordinate.js]: Yes, good catch! I'm also wondering if this deprecated approach should be changed to: `var renderWhenInformation = data.val().whereAddress || data.val().whereDescription`

This is assuming the 'Description' variables should get overwritten by more recent or specific time/location information.

**Comment by omarcodex** [comment]
@JohannBillar sounds good, let's walk through when you get here! In the mean time, we're addressing updates to Profile.

File: scripts/coordinate.js (hunk 1 above)
Comment (omarcodex) [reply scripts/coordinate.js]: Added--thanks! 👍

**Pull request merged**